### PR TITLE
fix: adjust feature detection for node 18.19 loader changes

### LINKED ISSUE #421: `tsx` doesn't work on Node 18.19
### Precheck

- [X] I searched existing issues before opening this one to avoid duplicates
- [X] I'm able to reproduce this issue and prove it with a minimal reproduction
- [X] I understand this is not a place to ask for free debugging support

### Problem

Unable to run `tsx` on Node 18:

```
❯ pnpm tsx ./foo.ts         

node:internal/process/esm_loader:40
      internalBinding('errors').triggerUncaughtException(
                                ^
Error: tsx must be loaded with --import instead of --loader
The --loader flag was deprecated in Node v20.6.0
    at Q (file:///Users/divyansh/foo/node_modules/.pnpm/tsx@4.6.0/node_modules/tsx/dist/esm/index.mjs:1:1793)
    at Hooks.addCustomLoader (node:internal/modules/esm/hooks:202:24)
    at Hooks.register (node:internal/modules/esm/hooks:168:16)
    at async initializeHooks (node:internal/modules/esm/utils:167:5)
    at async customizedModuleWorker (node:internal/modules/esm/worker:104:24)

Node.js v18.19.0
```

Working fine till Node v18.18 (https://nodejs.org/en/blog/release/v18.19.0#esm-and-customization-hook-changes)

### Minimal reproduction

```
// foo.ts:

console.log('hello')

// on terminal:

nvm use 18
pnpm add -D tsx
pnpm tsx ./foo.ts
```

### Expected behavior

Not fail on LTS versions.

### Version

v4.6.0

### Node.js version

v18.19.0

### Package manager

pnpm

### Operating system

macOS

### Contributions

- [x] I plan to open a pull request for this issue
- [ ] I plan to make a financial contribution to this project

### PR DESCRIPTION
**Pull request by brc-dd**
fixes #421

### CODE REVIEW

File: src/esm/loaders.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ type resolve = (
 
 export const initialize: InitializeHook = async (data) => {
 	if (!data) {
-		throw new Error('tsx must be loaded with --import instead of --loader\nThe --loader flag was deprecated in Node v20.6.0');
+		throw new Error('tsx must be loaded with --import instead of --loader\nThe --loader flag was deprecated in Node v20.6.0 and v18.19.0');
```
Comment (brc-dd) [inline src/esm/loaders.ts:42]: adjust this maybe 👀

File: tests/utils/node-versions.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export const nodeVersions = [
 		)
 			? [
 				latestMajor('20.10.0'),
-				latestMajor('18.18.2'),
+				latestMajor('18.19.0'),
```
Comment (brc-dd) [inline tests/utils/node-versions.ts:17]: should we add '18.8.2' here too?

File: src/esm/loaders.ts (hunk 1 above)
Comment (privatenumber) [reply src/esm/loaders.ts:42]: Looks good!

File: tests/utils/node-versions.ts (hunk 2 above)
Comment (privatenumber) [reply tests/utils/node-versions.ts:19]: All good. Technically, tsx just supports the _latest_ LTS but it's nice to have tests to make sure baseline functionality.

**Comment by privatenumber** [comment]
Thank you @brc-dd! Perfect PR 🔥

**Comment by privatenumber** [comment]
:tada: This issue has been resolved in v4.6.1

If you appreciate this project, please consider [supporting this project by sponsoring](https://github.com/sponsors/privatenumber) :heart: :pray:

**Pull request merged**